JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -177,7 +143,7 @@ internal async Task ListenLoop(CancellationToken ct)
                     {
                         foreach (var message in sqsMessageResponse.Messages)
                         {
-                            HandleMessage(message, ct);
+                            await HandleMessage(message, ct).ConfigureAwait(false);

```

Comment:
Something that I think was missed with my previous PR but we should check the cancellation token before calling `HandleMessage(...)` in the message loop

Reply:
So if the worker gets 2 messages off the queue, Handles 1 of them and then notices that `ct.IsCancellationRequested` is true (e.g. during scale down), it will not process the second one.  So then the message is retrieved but not deleted, and will soon time out and end up back on the queue for other worker to  take?

Reply:
Exactly correct. The other option is to try to commit to whatever messages we've peeked at from the queue, not sure which is the best option. Your thoughts?